Mathlib/Analysis/Normed/Module/Milman-Pettis.lean

```diff
@@ -0,0 +1,252 @@
+        (by fun_prop) <| WeakBilin.eval_continuous (strongDualPairing ℝ (StrongDual ℝ E)) _
+  refine ⟨V, hV_open, by simpa, ?_⟩
+  obtain ⟨ϖ, hϖ⟩ : (V ∩ 𝒰).Nonempty := by
+    rw [← closure_nonempty_iff (X := WeakDual ℝ (StrongDual ℝ E))]
+    exact ⟨ξ, hξ_mem hφ_lt hV_open⟩
+  use ϖ
+  intro ζ hζ
+  have {θ : E**} (hθ : θ ∈ V) : θ φ > 1 - δ'/2 := by
+    rw [hV_def, Set.mem_setOf_eq, abs_sub_lt_iff] at hθ
+    linarith
+  rw [mem_closedBall_iff_norm]
+  apply le_of_lt
+  obtain ⟨z, hz_norm, hzζ⟩ := hζ.2
+  obtain ⟨y, hy_norm, hyϖ⟩ := hϖ.2
+  rw [← hzζ, ← hyϖ, ← map_sub, double_dual_norm_eq]
+  apply h_UniformConvex (by rwa [← mem_closedBall_zero_iff]) (by rwa [← mem_closedBall_zero_iff])
+  rw [← double_dual_norm_eq ℝ, map_add, hzζ, hyϖ]
+  calc ‖ζ + ϖ‖ ≥ ‖(ζ + ϖ) φ‖ := by grw [(le_opNorm (f := ζ + ϖ) φ), hφ_norm, mul_one]
+             _ = |ζ φ + ϖ φ| := by simp
+             _ ≥ ζ φ + ϖ φ := le_abs_self ..
+             _ > 1 - δ'/2 + 1 - δ'/2 := by linarith [this hζ.1, this hϖ.1]
+             _ = 2 - δ' := by linarith
+
+
+lemma sphere_subset_closure [UniformConvexSpace E] : sphere 0 1 ⊆ closure 𝒰 := by
+  let 𝒯 : TopologicalSpace <| WeakDual ℝ (StrongDual ℝ E) := inferInstance
+  intro x hx
+  rw [Metric.mem_closure_iff]
+  intro ε hε
+  obtain ⟨W, hW, x_mem, c, hW_sub⟩ := exists_ball_lt (ε := ε/3) (by positivity) hx
+  have hx_mem : x ∈ closure[𝒯] (W ∩ 𝒰) := by
+    apply hW.inter_closure <| Set.mem_inter x_mem _
+    rw [goldstine]
+    exact sphere_subset_closedBall (α := E**) hx
+  obtain ⟨y, hy_mem⟩ : (W ∩ 𝒰).Nonempty := by
+    rw [← closure_nonempty_iff (X := WeakDual ℝ (StrongDual ℝ E))]
+    use x
+  refine ⟨y, hy_mem.2, ?_⟩
+  suffices x ∈ closedBall c (ε/2) by
+    apply lt_of_le_of_lt
+    apply dist_triangle  (y := c)
+    simp at this
+    grw [hW_sub] at hy_mem
+    simp only [mem_closedBall, dist_comm] at hy_mem
+    grw [hy_mem, this]
+    linarith
+  apply WeakClosure_subset_closedBall _ hx_mem
+  apply subset_trans hW_sub <| closedBall_subset_closedBall (by linarith)
+
+lemma sphere_subset_image [CompleteSpace E] [UniformConvexSpace E] : sphere 0 1 ⊆ 𝒰 := by
+  grw [sphere_subset_closure, IsClosed.closure_eq]
+  exact IsClosed_image_ball
+
+variable (E) [CompleteSpace E] [UniformConvexSpace E]
+
+/- Milman-Pettis theorem: every uniformly convex Banach (**FAE: Complete Needed?**) space is
+reflexive, stated as the surjectivity of `inclusionInDoubleDual`. For the version proving
+this is a linear isometric equivalence, see `LinearIsometryEquiv_of_uniformConvexSpace`. -/
+theorem surjective_of_uniformConvexSpace : Surjective (inclusionInDoubleDual ℝ E) :=
+  (surjective_iff_sphere_subset_range _).mpr
+    ⟨_, zero_lt_one, sphere_subset_image.trans <| Set.image_subset_range ..⟩
+
+/-- Milman-Pettis theorem: every uniformly convex Banach (**FAE: Complete Needed?**) space is
+reflexive. For a version proving only surjectivity, see `surjective_of_uniformConvexSpace`. -/
```

Comment:
Just having a quick look: completeness will be needed to get this statement at least, because `E**` is always complete. In the non-complete case you will get an isometry with dense range (by applying Milman-Pettis to the completion)